### PR TITLE
Feature/e UI 6516 task permission wa4

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-6516

### Change description ###
Fine grained task permissions for release 4 (turned on by LaunchDarkly feature to keep existing functionality)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: api/workAllocation/utils.spec.ts [hunk 1]
```diff
@@ -824,6 +825,53 @@ describe('workAllocation.utils', () => {
...
+      expect(getActionsByRefinedPermissions('ActiveTasksAssignedCurrentUser', [TaskPermission.UNASSIGNASSIGN, TaskPermission.CANCEL]))
+        .to.deep.equal([CANCEL, GO, REASSIGN]);
+      // EUI-5046 - ensure test includes check that own gives correct actions as well
+      expect(getActionsByRefinedPermissions('ActiveTasksAssignedCurrentUser', [TaskPermission.UNCLAIM, TaskPermission.CLAIM, TaskPermission.COMPLETEOWN, TaskPermission.OWN]))
+        .to.deep.equal([COMPLETE, GO, RELEASE]);
+      // ensure that in unlikely scenario of below that no duplication occurs
+      expect(getActionsByRefinedPermissions('ActiveTasksAssignedOtherUser', [TaskPermission.UNASSIGN, TaskPermission.ASSIGN, TaskPermission.OWN]))
+        .to.deep.equal([CLAIM, GO, REASSIGN, RELEASE]);
+    });
+
```
Comment (MateiHMCTS) [inline api/workAllocation/utils.spec.ts:872]: styling issue

File: api/configuration/references.ts [hunk 2]
```diff
@@ -111,6 +111,8 @@ export const FEATURE_ACCESS_MANAGEMENT_ENABLED = 'accessManagementEnabled';
 
 export const FEATURE_WORKALLOCATION_ENABLED = 'workAllocationEnabled';
 
+export const FEATURE_UPDATED_TASK_PERMISSIONS_ENABLED = 'updatedTaskPermissionsEnabled';
+
```
Comment (MateiHMCTS) [inline api/configuration/references.ts:115]: styling issue

File: src/work-allocation/services/work-allocation-task.service.ts [hunk 3]
```diff
@@ -54,7 +54,7 @@ export class WorkAllocationTaskService {
     return this.http.post<any>(`${BASE_URL}`, task);
```
Comment (MateiHMCTS) [inline src/work-allocation/services/work-allocation-task.service.ts:54]: why any?

File: src/work-allocation/services/work-allocation-task.service.ts [hunk 4]
```diff
@@ -54,7 +54,7 @@ export class WorkAllocationTaskService {
...
   }
 
-  public searchTask(body: { searchRequest: SearchTaskRequest, view: string }): Observable<TaskResponse> {
+  public searchTask(body: { searchRequest: SearchTaskRequest, view: string, refined: boolean }): Observable<TaskResponse> {
```
Comment (MateiHMCTS) [inline src/work-allocation/services/work-allocation-task.service.ts:57]: why pass refined if not used?

**Review by MateiHMCTS — CHANGES_REQUESTED** [review]

**Review by Surekha-HMCTS — APPROVED** [review]

**Review by DavidJayakumar — APPROVED** [review]
Looks good

**Review by Kartheek-S — APPROVED** [review]

**Pull request merged**